lib/src/contracts/abi/abi.dart

```diff
@@ -218,18 +218,34 @@ class ContractFunction {
   /// * uint<x> and int<x> will accept a dart int
   ///
   /// Other types are not supported at the moment.
-  Uint8List encodeCall(List<dynamic> params) {
-    if (params.length != parameters.length) {
-      throw ArgumentError.value(
-          params.length, 'params', 'Must match function parameters');
-    }
+  Uint8List encodeCall([List<dynamic> params]) {
```

Comment:
I don't think `params` should ever be null. If you want to make it optional, let's add a default parameter:

```suggestion
  Uint8List encodeCall([List<dynamic> params = const []]) {
```

Reply:
Well, I think so. So I changed it.